### PR TITLE
Include nexus-iq-server-ha helm chart in to the release - CLM-23674

### PR DESCRIPTION
**Pull request by kakumara**
#### Description of Change

This includes the new nexus-iq-server-HA helm chart in to the chart release cycle. The chart is currently hosted/maintained in https://github.com/sonatype/nexus-iq-server-ha/tree/main/chart. 

**NOTE:**
The repository above is still not public (we will be making it public along with the official HA release). So until then, plan is to keep this PR as a DRAFT (not to publish the chart yet to artifacthub)

### CODE REVIEW

File: Jenkinsfile-Release [hunk 1]
```diff
@@ -63,6 +65,11 @@ dockerizedBuildPipeline(
       runSafely 'git add charts'
     }
 
+    if ('nexus-iq-server-ha'.equals(params.chart)) {
+      runSafely "curl -L 'https://github.com/sonatype/nexus-iq-server-ha/blob/main/docs/${nexusIqServerHaArtifactName}?raw=true' -o 'docs/${nexusIqServerHaArtifactName}'"
+      runSafely './build-nexus.sh'
```
Comment (mealingr) [inline Jenkinsfile-Release:70]: Was curious if running `./build-nexus.sh` is correct but it's just running
```
cd docs
helm repo index . --url https://sonatype.github.io/helm3-charts
```
which should be fine for the IQ HA chart too

**Review by mealingr — APPROVED** [review]
+1

**Review by mr-zepol — APPROVED** [review]
+1 , also note the comment about the release process we discussed this morning, if this new value is automatically included in the release we should not merge yet, but if we need to update something else maybe worth creating a ticket to follow up on that

**Comment by kakumara** [comment]
> +1 , also note the comment about the release process we discussed this morning, if this new value is automatically included in the release we should not merge yet, but if we need to update something else maybe worth creating a ticket to follow up on that

Fair point :+1: . The `helm3-charts` release is triggered as a post release downstream task here. https://github.com/sonatype/insight-brain/blob/main/Jenkinsfile.postRelease#L78 it is currently set to pick only the `nexus-iq-server` option. So we should be safe. 
But that would require us to update the the post release job later on (again) so I changed this script to trigger both charts to be released jut from the `nexus-iq-server` option. https://github.com/sonatype/helm3-charts/pull/255/commits/2817a89d366fecd68870929f8dc5cfee64d07ce0 (So changing the PR back to draft until we are ready to release this)

File: Jenkinsfile-Release (hunk 1 above)
Comment (jflinchbaugh) [reply Jenkinsfile-Release:66]: it may be time to rename that `build-nexus.sh` to `build-index.sh` or something, since it's taken on a generic use.

Comment (jflinchbaugh) [reply Jenkinsfile-Release:66]: alternatively, you could `curl` the HA chart first, then `upgrade.sh` and `build-iq.sh` and let that build the index once.

**Comment by mr-zepol** [comment]
> > +1 , also note the comment about the release process we discussed this morning, if this new value is automatically included in the release we should not merge yet, but if we need to update something else maybe worth creating a ticket to follow up on that
> 
> Fair point 👍 . The `helm3-charts` release is triggered as a post release downstream task here. https://github.com/sonatype/insight-brain/blob/main/Jenkinsfile.postRelease#L78 it is currently set to pick only the `nexus-iq-server` option. So we should be safe. But that would require us to update the the post release job later on (again) so I changed this script to trigger both charts to be released jut from the `nexus-iq-server` option. [2817a89](https://github.com/sonatype/helm3-charts/commit/2817a89d366fecd68870929f8dc5cfee64d07ce0) (So changing the PR back to draft until we are ready to release this)

if needed ( in order to test) we can leave as it was and then create a new ticket to include this in the post-release job though

**Comment by kakumara** [comment]
> if needed ( in order to test) we can leave as it was and then create a new ticket to include this in the post-release job though

I've added the remaining work to be done in [CLM-23682](https://issues.sonatype.org/browse/CLM-23682)

File: Jenkinsfile-Release (hunk 1 above)
Comment (kakumara) [reply Jenkinsfile-Release]: > alternatively, you could `curl` the HA chart first, then `upgrade.sh` and `build-iq.sh` and let that script build the index once.

:+1: I adopted this option 960168b

**Review by jflinchbaugh — APPROVED** [review]
+1

**Pull request merged**